blueprints/cloud-operations/network-dashboard/cloud-function/metrics/subnets.py

```diff
@@ -0,0 +1,251 @@
+          "read_mask": read_mask,
+      })
+
+  # Counting IP addresses for GCE Internal Load Balancers
+  for asset in response_ilb:
+    internal = False
+    network_name = ''
+    project_id = ''
+    subnet_name = ''
+    subnet_region = ''
+    for versioned in asset.versioned_resources:
+      for field_name, field_value in versioned.resource.items():
+        if 'loadBalancingScheme' in field_name and field_value == 'INTERNAL':
+          internal = True
+        if field_name == 'network':
+          network_name = field_value.split('/')[-1]
+          project_id = field_value.split('/')[6]
+        if 'subnetwork' in field_name:
+          subnet_name = field_value.split('/')[-1]
+          subnet_region = field_value.split('/')[-3]
+
+    if internal:
+      all_subnets_dict[project_id][f"{subnet_region}/{subnet_name}"][
+          'used_ip_addresses'] += 1
+
+  response_reserved_ips = config["clients"][
+      "asset_client"].search_all_resources(
+          request={
+              "scope": f"organizations/{config['organization']}",
+              "asset_types": ["compute.googleapis.com/Address"],
+              "read_mask": read_mask,
+          })
+
+  # Counting IP addresses for GCE Reserved IPs (ex: PSC, Cloud DNS Inbound policies, reserved GCE IPs)
+  for asset in response_reserved_ips:
+    purpose = ""
+    status = ""
+    project_id = ""
+    network_name = ""
+    subnet_name = ""
+    subnet_region = ""
+    address = ""
+    prefixLength = ""
+    for versioned in asset.versioned_resources:
+      for field_name, field_value in versioned.resource.items():
+        if field_name == 'purpose':
+          purpose = field_value
+        if field_name == 'region':
+          subnet_region = field_value.split('/')[-1]
+        if field_name == 'status':
+          status = field_value
+        if field_name == 'address':
+          address = field_value
+        if field_name == 'network':
+          network_name = field_value.split('/')[-1]
+          project_id = field_value.split('/')[6]
+        if field_name == 'subnetwork':
+          subnet_name = field_value.split('/')[-1]
+          # Cloud DNS doesn't include the 'network' or 'project' info so we get it here
+          project_id = field_value.split('/')[6]
+        if field_name == 'prefixLength':
+          prefixLength = field_value
+
+    # GCE instance or PSC Forwarding Rule
+    if purpose == "GCE_ENDPOINT" and status == "RESERVED":
+      all_subnets_dict[project_id][f"{subnet_region}/{subnet_name}"][
+          'used_ip_addresses'] += 1
+    # Cloud DNS inbound policy
+    if purpose == "DNS_RESOLVER":
+      all_subnets_dict[project_id][f"{subnet_region}/{subnet_name}"][
+          'used_ip_addresses'] += 1
+    # PSA Range for Cloud SQL, MemoryStore, etc.
+    if purpose == "VPC_PEERING":
+      # TODO: PSA range to be handled later
+      # print("PSA range to be handled later:", address, prefixLength, network_name)
+      continue
```

Comment:
again elif could probably improve the speed a bit. 
I tested locally and saw an improvement of ~40% when item 1/3 was matching. And an equal performance for the last item